source/OutfitInfoDisplay.cpp

```diff
@@ -171,147 +171,117 @@ void OutfitInfoDisplay::UpdateAttributes(const Outfit &outfit)
 	attributeValues.clear();
 	attributesHeight = 20;
 	
-	map<string, map<string, int>> listing;
-	for(const auto &it : outfit.Attributes())
+	for(const pair<string, double> &it : outfit.Attributes())
```

Comment:
When you do this, what's actually happening is that you're constructing a copy of each attributes node (via pair's templated constructor) and then getting a const reference to that temporary (which extends its lifetime) and then using that reference. That's a lot of behind-the-scenes stuff to hide behind a for loop. I think it'd make more sense to just iterate through the actual items in the attributes set (which are pair<const char *, double>s) and then typecast the name to a string if needed for concatenating strings.

Reply:
That was how I had it originally, and then when I changed it to test if there would have been a warning when the type of `it` changed from <string, double> to <const char *, double> I found that no warning was generated (and no display bug from adding `':'` to the char * was present either), so I stuck with it.
I'll swap back